File: app/src/main/java/me/imzack/app/ender/injector/component/TypeCreationComponent.java
package me.imzack.app.ender.injector.component;

import me.imzack.app.ender.injector.module.TypeCreationPresenterModule;
import me.imzack.app.ender.injector.scope.ActivityScope;
import me.imzack.app.ender.view.activity.TypeCreationActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = TypeCreationPresenterModule.class)
public interface TypeCreationComponent {

    void inject(TypeCreationActivity typeCreationActivity);
}

File: app/src/main/java/net/zackzhang/app/end/injector/component/TypeCreationComponent.kt
package net.zackzhang.app.end.injector.component

import dagger.Component
import net.zackzhang.app.end.injector.module.TypeCreationPresenterModule
import net.zackzhang.app.end.injector.scope.ActivityScope
import net.zackzhang.app.end.view.activity.TypeCreationActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(TypeCreationPresenterModule::class))
interface TypeCreationComponent {

    fun inject(typeCreationActivity: TypeCreationActivity)
}
